### ISSUE TITLE
Elixir 1.17 complains about unimplemented behaviours

### ISSUE THREAD

**Issue by AndrewDryga** [elixir-cldr/cldr_calendars#19]
Versions:
```
❯ mix deps | grep ex_cldr         
* ex_cldr (Hex package) (mix)
  locked at 2.39.1 (ex_cldr) e983f6f8
* ex_cldr_calendars 1.23.1 (Hex package) (mix)
  locked at 1.23.1 (ex_cldr_calendars) 2983f002
* ex_cldr_currencies (Hex package) (mix)
  locked at 2.16.1 (ex_cldr_currencies) 095d5e97
* ex_cldr_dates_times (Hex package) (mix)
  locked at 2.18.0 (ex_cldr_dates_times) a8fbe9e2
* ex_cldr_numbers (Hex package) (mix)
  locked at 2.33.1 (ex_cldr_numbers) c003bfaa
```

Warnings:
```
mix deps.compile ex_cldr_calendars
==> ex_cldr_calendars
Compiling 25 files (.ex)
    warning: function shift_date/4 required by behaviour Calendar is not implemented (in module Cldr.Calendar.Gregorian)
    │
  3 │ defmodule Cldr.Calendar.Gregorian do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/gregorian.ex:3: Cldr.Calendar.Gregorian (module)

    warning: function shift_date/4 required by behaviour Calendar is not implemented (in module Cldr.Calendar.NRF)
    │
  3 │ defmodule Cldr.Calendar.NRF do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/nrf.ex:3: Cldr.Calendar.NRF (module)

    warning: function shift_date/4 required by behaviour Calendar is not implemented (in module Cldr.Calendar.Julian)
    │
  1 │ defmodule Cldr.Calendar.Julian do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/julian.ex:1: Cldr.Calendar.Julian (module)

    warning: function shift_date/4 required by behaviour Calendar is not implemented (in module Cldr.Calendar.ISOWeek)
    │
  3 │ defmodule Cldr.Calendar.ISOWeek do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/iso_week.ex:3: Cldr.Calendar.ISOWeek (module)

    warning: function shift_date/4 required by behaviour Calendar is not implemented (in module Cldr.Calendar.ISO)
    │
  3 │ defmodule Cldr.Calendar.ISO do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/iso.ex:3: Cldr.Calendar.ISO (module)

    warning: function shift_naive_datetime/8 required by behaviour Calendar is not implemented (in module Cldr.Calendar.NRF)
    │
  3 │ defmodule Cldr.Calendar.NRF do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/nrf.ex:3: Cldr.Calendar.NRF (module)

    warning: function shift_naive_datetime/8 required by behaviour Calendar is not implemented (in module Cldr.Calendar.Gregorian)
    │
  3 │ defmodule Cldr.Calendar.Gregorian do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/gregorian.ex:3: Cldr.Calendar.Gregorian (module)

    warning: function shift_naive_datetime/8 required by behaviour Calendar is not implemented (in module Cldr.Calendar.ISOWeek)
    │
  3 │ defmodule Cldr.Calendar.ISOWeek do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/iso_week.ex:3: Cldr.Calendar.ISOWeek (module)

    warning: function shift_naive_datetime/8 required by behaviour Calendar is not implemented (in module Cldr.Calendar.Julian)
    │
  1 │ defmodule Cldr.Calendar.Julian do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/julian.ex:1: Cldr.Calendar.Julian (module)

    warning: function shift_naive_datetime/8 required by behaviour Calendar is not implemented (in module Cldr.Calendar.ISO)
    │
  3 │ defmodule Cldr.Calendar.ISO do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/iso.ex:3: Cldr.Calendar.ISO (module)

    warning: function shift_time/5 required by behaviour Calendar is not implemented (in module Cldr.Calendar.NRF)
    │
  3 │ defmodule Cldr.Calendar.NRF do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/nrf.ex:3: Cldr.Calendar.NRF (module)

    warning: function shift_time/5 required by behaviour Calendar is not implemented (in module Cldr.Calendar.Gregorian)
    │
  3 │ defmodule Cldr.Calendar.Gregorian do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/gregorian.ex:3: Cldr.Calendar.Gregorian (module)

    warning: function shift_time/5 required by behaviour Calendar is not implemented (in module Cldr.Calendar.ISOWeek)
    │
  3 │ defmodule Cldr.Calendar.ISOWeek do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/iso_week.ex:3: Cldr.Calendar.ISOWeek (module)

    warning: function shift_time/5 required by behaviour Calendar is not implemented (in module Cldr.Calendar.Julian)
    │
  1 │ defmodule Cldr.Calendar.Julian do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/julian.ex:1: Cldr.Calendar.Julian (module)

    warning: function shift_time/5 required by behaviour Calendar is not implemented (in module Cldr.Calendar.ISO)
    │
  3 │ defmodule Cldr.Calendar.ISO do
    │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    │
    └─ lib/cldr/calendar/calendars/iso.ex:3: Cldr.Calendar.ISO (module)

Generated ex_cldr_calendars app
```

**Comment by AndrewDryga** [comment]
Also, wanted to thank you for this all the CLDR packages, you are a monster, in a good way :).

**Comment by kipcole9** [comment]
@AndrewDryga, thanks for the issue. I'm a bit embarrassed I haven't got these callbacks implemented in time for Elixir 1.17 official release. First time I've missed a compatibility in a loooong time. 

I have done quite a bit of work on this, but it's not finished and not tested. I'll try hard to get a release out over the weekend.

**Comment by AndrewDryga** [comment]
@kipcole9 there is nothing to be embarrassed of, we all have life and work, and you can't maintain open source 24/7, especially having as many packages as you ❤️.

**Comment by kipcole9** [comment]
I've pushed [a commit](https://github.com/elixir-cldr/cldr_calendars/commit/57fef27292ec1965b6da4d1ddeb06455b900459f) that implements the new callbacks. Using the same tests from Elixir, but using `Cldr.Calendar.Gregorian` passes. 

I think this is ready for some more testing if you care to configure `{:ex_cldr_calendars, github: "elixir-cldr/cldr_calendars"}`.

**Comment by kipcole9** [comment]
I've published [ex_cldr_calendars version 1.24.0](https://hex.pm/packages/ex_cldr_calendars/1.24.0) with the following changelog entry:

### Bug Fixes

* Fix compiler warnings for Elixir 1.17.

### Enhancements

* Adds new calendar callbacks `Calendar.shift_date/4`, `Calendar.shift_time/5` and `Calendar.shift_naive_datetime/8` supported in Elixir 1.17.

Of course please re-open if you see anything incorrect or unexpected.

**Comment by AndrewDryga** [comment]
You were so fast I wasn't able to test it, but everything looks good. Thank you!
